MonstieWash/Assets/Scripts/Tutorial/TutorialManager.cs

```diff
@@ -199,95 +195,94 @@ private void TimeCompletion(float targetValue)
     #endregion
 
     #region Event Listeners
-    private void OnMove(Vector2 movement)
+
+    private void EventTriggered(CompletionEvent eventType)
     {
         var currentPrompt = m_tutorialPrompts[m_tutorialStep];
-        if (currentPrompt.CompleteEvent == CompletionEvent.OnMove)
+        if (currentPrompt.CompleteEvent == eventType)
         {
             RunCompletionTests(currentPrompt);
         }
     }
 
-    private void EraseStart(bool value)
+    private void OnMove(Vector2 movement)
     {
-        var currentPrompt = m_tutorialPrompts[m_tutorialStep];
-        if (currentPrompt.CompleteEvent == CompletionEvent.EraseStarted)
+        EventTriggered(CompletionEvent.OnMove);
+    }
+
+    private void EraseStart(bool value, Tool tool)
+    {
+        switch (tool.TypeOfTool)
         {
-            RunCompletionTests(currentPrompt);
+            case Tool.ToolType.Brush:
+                Debug.Log("Use brush");
+                EventTriggered(CompletionEvent.UseBrush);
+                break;
+            case Tool.ToolType.Sponge:
+                Debug.Log("Use sponge");
+                EventTriggered(CompletionEvent.UseSponge);
+                break;
+            case Tool.ToolType.WaterWand:
+                Debug.Log("Use water wand");
+                EventTriggered(CompletionEvent.UseWaterWand);
```

Comment:
**Cleanup** | _This is for keeping the codebase clean and manageable._

With these, just make sure to remove needless debug logs once you're done testing or at least lock them behind a `showDebug` flag.

Reply:
damn it i always forget to remove these